src/Services/M3U.vala

```diff
@@ -0,0 +1,69 @@
+/*
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ * SPDX-FileCopyrightText: 2025 elementary, Inc. (https://elementary.io)
+ */
+
+namespace Music.M3U {
+
+    // Standard specification here: https://en.wikipedia.org/wiki/M3U
+    public File[]? parse_playlist (File playlist) {
+        debug ("Parsing playlist: %s", playlist.get_path ());
+        File[] list = {};
+
+        try {
+            FileInputStream @is = playlist.read ();
+            DataInputStream dis = new DataInputStream (@is);
+            string line;
+
+            while ((line = dis.read_line ()) != null) {
+                debug ("%s", line);
+
+                // Skip extended
```

Comment:
```suggestion
                // Skip extended
```

Generally outputting file content without validation in a log message is potentially unsafe. We don't have any validations here so I don't think it's guaranteed as ASCII text